File: backend/buildeR.DAL/Entities/TeamMember.cs
using buildeR.Common.Enums;
using buildeR.DAL.Entities.Common;
using System;

namespace buildeR.DAL.Entities
{
    public class TeamMember: Entity
    {
        public int UserId { get; set; }
        public GroupRole MemberRole { get; set; }
        public int GroupId { get; set; }
        public DateTime JoinedDate { get; set; }
        public bool IsAccepted { get; set; }
        public virtual Group Group { get; set; }
        public virtual User User { get; set; }
    }
}

File: frontend/src/app/shared/models/group/team-member.ts
import { User } from '../user/user';
import { Group } from './group';
import { GroupRole } from './group-role';

export interface TeamMember {
  id: number;
  userId: number;
  groupId: number;
  memberRole: GroupRole;
  user: User;
  group: Group;
  joinedDate: Date;
  isAccepted: boolean;
}
